infrastructure/lib/atat-web-api-stack.ts

```diff
@@ -34,125 +34,91 @@ export class AtatWebApiStack extends cdk.Stack {
-    // OperationIds from API spec are used to identify functions below
-
-    const createPortfolioDraft = new ApiDynamoDBFunction(this, "CreatePortfolioDraft", {
-      resource: portfolioDrafts,
+    const createPortfolioDraft = new ApiDynamoDBFunction(this, "CreatePortfolioDraftFunction", {
       table: table,
       method: HttpMethod.POST,
       handlerPath: packageRoot() + "/api/portfolioDrafts/createPortfolioDraft.ts",
     });
 
-    const getPortfolioDrafts = new ApiDynamoDBFunction(this, "GetPortfolioDrafts", {
-      resource: portfolioDrafts,
+    const getPortfolioDrafts = new ApiDynamoDBFunction(this, "GetPortfolioDraftsFunction", {
       table: table,
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/getPortfolioDrafts.ts",
     });
 
-    const deletePortfolioDraft = new ApiDynamoDBFunction(this, "DeletePortfolioDraft", {
-      resource: portfolioDraftId,
+    const deletePortfolioDraft = new ApiDynamoDBFunction(this, "DeletePortfolioDraftFunction", {
       table: table,
       method: HttpMethod.DELETE,
       handlerPath: packageRoot() + "/api/portfolioDrafts/deletePortfolioDraft.ts",
     });
 
-    const createPortfolioStep = new ApiDynamoDBFunction(this, "CreatePortfolioStep", {
-      resource: portfolio,
-      table: table,
-      method: HttpMethod.POST,
-      handlerPath: packageRoot() + "/api/portfolioDrafts/portfolio/createPortfolioStep.ts",
-    });
-
-    const getPortfolioStep = new ApiDynamoDBFunction(this, "GetPortfolioStep", {
-      resource: portfolio,
+    const getPortfolioStep = new ApiDynamoDBFunction(this, "GetPortfolioStepFunction", {
       table: table,
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/portfolio/getPortfolioStep.ts",
     });
 
-    const createFundingStep = new ApiDynamoDBFunction(this, "CreateFundingStep", {
-      resource: funding,
+    const createFundingStep = new ApiDynamoDBFunction(this, "CreateFundingStepFunction", {
       table: table,
       method: HttpMethod.POST,
       handlerPath: packageRoot() + "/api/portfolioDrafts/funding/createFundingStep.ts",
     });
 
-    const getFundingStep = new ApiDynamoDBFunction(this, "GetFundingStep", {
-      resource: funding,
+    const getFundingStep = new ApiDynamoDBFunction(this, "GetFundingStepFunction", {
       table: table,
       method: HttpMethod.GET,
       handlerPath: packageRoot() + "/api/portfolioDrafts/funding/getFundingStep.ts",
     });
 
+    // The API spec, which just so happens to be a valid CloudFormation snippet (with some actual CloudFormation
+    // in it) gets uploaded to S3
+    const apiAsset = new s3asset.Asset(this, "ApiSpecAsset", {
+      path: "./atat_provisioning_wizard_api.yaml",
+    });
+
+    // And now we include that snippet as an actual part of the template using the AWS::Include Transform. This
+    // is where the two previous pieces come together. Now, all the Fn::Sub and other functions will be resolved
```

Comment:
I think we may need a small explanation here for the uninitiated among us. If the first piece is the spec in a S3 bucket, what's the other previous piece? The overall goal here is to connect the Lambdas to the APIGW Routes right?

Reply:
Those comments are a bit confusing because I incorporated the logicalId into the api-function construct. We don't need to define the lambda integration because we are linking to it in the `uri` section of the `x-amazon-api-gateway-integration`.

Reply:
Before there were 2 extra lines, extracting the logicalId so it could play nice with Cloudformation, but this logic was incorporated into the api-function construct to avoid repetition.

Reply:
I'm still a bit lost there. We can take this conversation to standup and hopefully give us the context we need (and possibly update these comments but I don't want to slow down the trains).

Reply:
I think we just need to rewrite the flow of these comments. Let me know when you're done with the merge work and I'll work on that since I wrote the bad comment in the first place.